File: bubble.java
import java.io.*;
import java.nio.file.*;
import java.util.ArrayList;
import java.util.List;

public class bubble {

    // Método para realizar o Bubble Sort
    public static void bubbleSort(List<Integer> arr) {
        int n = arr.size();
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr.get(j) > arr.get(j + 1)) {
                    // Troca de elementos
                    int temp = arr.get(j);
                    arr.set(j, arr.get(j + 1));
                    arr.set(j + 1, temp);
                }
            }
        }
    }

    // Método principal para medir tempo e executar a lógica
    public static void measureMemoryAndTime(String filePath) throws IOException {
        // Ler números do arquivo
        List<Integer> numbers = new ArrayList<>();
        for (String line : Files.readAllLines(Paths.get(filePath))) {
            numbers.add(Integer.parseInt(line.trim()));
        }

        // Medir tempo inicial
        long startTime = System.nanoTime();

        // Realizar o Bubble Sort
        bubbleSort(numbers);

        // Medir tempo final
        long endTime = System.nanoTime();
        double executionTimeMs = (endTime - startTime) / 1_000_000.0;

        // Escrever números ordenados em um arquivo
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"))) {
            for (int num : numbers) {
                writer.write(num + "\n");
            }
        }

        // Obter o uso de memória
        long memoryUsageKb = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / 1024;

        // Exibir resultados
        System.out.printf("Tempo de execução: %.2f ms%n", executionTimeMs);
        System.out.printf("Uso de memória: %d KB%n", memoryUsageKb);
    }

    public static void main(String[] args) {
        String inputFile = "input.txt";

        try {
            measureMemoryAndTime(inputFile);
        } catch (IOException e) {
            System.err.println("Erro ao processar o arquivo: " + e.getMessage());
        }
    }
}

File: bubble.py
import time
import psutil  # type: ignore
import os

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr

def measure_memory_and_time(file_path):
    # Verifica se o arquivo existe
    if not os.path.exists(file_path):
        raise FileNotFoundError(f"Arquivo não encontrado: {file_path}")

    # Leitura do arquivo
    with open(file_path, "r") as file:
        numbers = [int(line.strip()) for line in file]

    # Medir o tempo inicial
    start_time = time.time()

    # Algoritmo Bubble Sort
    sorted_numbers = bubble_sort(numbers)

    # Medir o tempo final
    end_time = time.time()
    execution_time_ms = (end_time - start_time) * 1000  # Em milissegundos

    # Gravar a saída ordenada
    with open("output.txt", "w") as file:
        for num in sorted_numbers:
            file.write(f"{num}\n")

    # Medir memória usada
    process = psutil.Process(os.getpid())
    memory_usage_kb = process.memory_info().rss / 1024  # Em KB

    return execution_time_ms, memory_usage_kb

if __name__ == "__main__":
    input_file = "input.txt"
    
    try:
        execution_time, memory_usage = measure_memory_and_time(input_file)
        print(f"Tempo de execução: {execution_time:.2f} ms")
        print(f"Uso de memória: {memory_usage:.2f} KB")
    except FileNotFoundError as e:
        print(e)
        print("Por favor, verifique se o arquivo input.txt está no local correto.")

